LunrCore/Extensions/DictionaryExtensions.cs

```diff
@@ -0,0 +1,48 @@
+﻿using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace Lunr;
+
+internal static class DictionaryExtensions
+{
+    public static void Increment(this Dictionary<string, int> dic, string fieldName, int amount = 1)
+    {
+#if NET6_0_OR_GREATER
+
+        ref int value = ref CollectionsMarshal.GetValueRefOrAddDefault(dic, fieldName, out _);
+
+        value += amount;
+#else
+        if (!dic.ContainsKey(fieldName))
+        {
+            dic.Add(fieldName, amount);
+        }
+        else 
+        {
+            dic[fieldName] += amount;
+        }
+#endif
+
+    }
+
+    public static void Increment(this Dictionary<string, double> dic, string fieldName, double amount = 1)
```

Comment:
Can you collapse those two into a single generic `Increment<T> where T:struct`?

Reply:
Was my initial goal, but we can't use += in generics until we get static abstracts (currently in preview) and the generic math interface.